tools/rbac-gen/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+var (
+	yamlFile = flag.String("yaml", "manifest.yaml", "yaml file from which the rbac will be generated.")
+	name = flag.String("name", "generated-role", "name of role to be generated")
+	saName = flag.String("sa-name", "", "name of service account the role should be binded to")
+	ns = flag.String("ns", "kube-system", "namespace of the role to be generated")
+	out = flag.String("out", "", "name of output file")
+	supervisory = flag.Bool("supervisory", false, "outputs role for operator in supervisory mode")
```

Comment:
Nit: we might want a string enum (as otherwise we tend to end up with lots of flags :-) )